### PR TITLE
IM measurement details

### PR DESCRIPTION
**Pull request by sarathms**
Shows the same details as shown in the mobile app.

### CODE REVIEW

File: renderer/components/nettests/im/FacebookMessenger.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    </Box>
+  )
+
+  return (
+    <div>
+      {render({
+        heroTitle: heroTitle,
+        details: <FBDetails />
+
+      })}
```
[the remark this answers is not in the document]
Comment (sarathms) [reply renderer/components/nettests/im/FacebookMessenger.js]: None. Will remove it.

File: renderer/components/nettests/im/Telegram.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  let appStatus = 'Okay'
+  let webStatus = 'Okay'
+  if (testKeys['telegram_http_blocking'] === true && testKeys['telegram_web_blocking'] === true) {
+    appStatus = 'Failed'
+    webStatus = 'Failed'
+  } else if (testKeys['telegram_http_blocking'] === true) {
+    appStatus = 'Failed'
+  } else if (testKeys['telegram_web_blocking'] === true) {
+    webStatus = 'Failed'
+  }
```
[the remark this answers is not in the document]
Comment (sarathms) [reply renderer/components/nettests/im/Telegram.js]: Earlier, we [had a text that explained](https://github.com/ooni/probe-desktop/blob/master/renderer/components/nettests/im/facebook-messenger.js#L73) the reason for the block which could be one of `{dns, tcp, dnsANDtcp}`. So we had to determine one of the three states. `probe-mobile` moved away from that and I am following it. This is a remnant of that which I [eventually changed for WhatsApp](https://github.com/ooni/probe-desktop/pull/57/files#diff-fb29a4d8f544f63891f81537bb57afbeR19-R27). When doing Telegram and Facebook Messenger, I wasn't sure if we wanted to drop that text altogether.

Now, I think it is better to apply the simpler one used in WhatsApp to both Telegram and FBM.

File: renderer/components/nettests/im/WhatsApp.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  Flex,
+  Box,
+  Text
+} from 'ooni-components'
+import { FormattedMessage } from 'react-intl'
+
+import StatusBox from '../../measurement/StatusBox'
+
+const WhatsApp = ({measurement, isAnomaly, render}) => {
+  // {"registration_server_blocking":false,"whatsapp_web_blocking":false,"whatsapp_endpoints_blocking":false}
```
[the remark this answers is not in the document]
Comment (sarathms) [reply renderer/components/nettests/im/WhatsApp.js]: Yes. I feel it is evident which test_keys are being used to determine. I had it for quick reference. Will remove.

File: renderer/components/nettests/im/Telegram.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (bassosimone) [reply renderer/components/nettests/im/Telegram.js]: ACK

File: renderer/components/nettests/im/Telegram.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+import StatusBox from '../../measurement/StatusBox'
+
+const Telegram = ({measurement, isAnomaly, render}) => {
+  const testKeys = JSON.parse(measurement.test_keys)
+  let appStatus = 'Okay'
+  let webStatus = 'Okay'
+
+  if (testKeys['telegram_http_blocking'] === true) {
+    appStatus = 'Failed'
```
Comment (bassosimone) [inline renderer/components/nettests/im/Telegram.js]: So, I think the app does not work if either `telegram_http_blocking` or `telegram_tcp_blocking`. Does it make sense to you? Or, am I misreading the spec https://github.com/ooni/spec/blob/master/nettests/ts-020-telegram.md?

Comment (sarathms) [reply renderer/components/nettests/im/Telegram.js]: You are right. I should be looking at `telegram_tcp_blocking` as well.

**Pull request merged**